pkg/simulate/pushsync/pushsync.go

```diff
@@ -0,0 +1,305 @@
+	DownloadWait        time.Duration
+	UploadWait          time.Duration
+	DownloadNodeAttemps int64
+	DownloadRetry       int64
+	Seed                int64
+	ProxyApiEndpoint    string
+	ChunkCount          int64
+	StartPercentage     float64
+	EndPercentage       float64
+	StepPercentage      float64
+}
+
+// NewDefaultOptions returns new default options
+func NewDefaultOptions() Options {
+	return Options{
+		GasPrice:            "",
+		MetricsPusher:       nil,
+		PostageAmount:       1000,
+		PostageDepth:        20,
+		PostageLabel:        "test-label",
+		UploadWait:          30 * time.Second,
+		DownloadWait:        15 * time.Second,
+		DownloadRetry:       3,
+		DownloadNodeAttemps: 3,
+		Seed:                0,
+		ProxyApiEndpoint:    "http://ethproxy.localhost",
+		ChunkCount:          500,
+		StartPercentage:     0.0,
+		EndPercentage:       0.6,
+		StepPercentage:      0.1,
+	}
+}
+
+// compile simulation whether Upload implements interface
+var _ beekeeper.Action = (*Simulation)(nil)
+
+// Simulation instance
+type Simulation struct{}
+
+// NewSimulation returns new upload simulation
+func NewSimulation() beekeeper.Action {
+	return &Simulation{}
+}
+
+// Run executes upload stress
+func (s *Simulation) Run(ctx context.Context, cluster *bee.Cluster, opts interface{}) (err error) {
+
+	o, ok := opts.(Options)
+	if !ok {
+		return fmt.Errorf("invalid options type")
+	}
+
+	metrics := newMetrics(cluster.Name()+"-"+time.Now().UTC().Format("2006-01-02-15-04-05-000000000"), o.MetricsPusher)
+
+	rnd := random.PseudoGenerator(o.Seed)
+
+	// begin recording block number
+	proxy := proxyClient.NewClient(o.ProxyApiEndpoint)
+	cancelID, err := proxy.Execute(proxyClient.BlockNumberRecord)
+	defer func(cancelID int) {
+		_ = proxy.Cancel(cancelID)
+	}(cancelID)
+
+	if err != nil {
+		return fmt.Errorf("proxy execute %w", err)
+	}
+	time.Sleep(time.Second * 5)
+
+	// set up clients
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return fmt.Errorf("node clients: %w", err)
+	}
+
+	if len(clients) <= 2 {
+		return errors.New("not enough nodes")
+	}
+
+	names := shuffle(rnd, cluster.NodeNames(), o.Seed)
+
+	//
+
+	buckets, leftovers := toBuckets(names, o.StartPercentage, o.EndPercentage, o.StepPercentage)
+
+	malfunctionEth := 0
+
+	for _, bucket := range buckets {
+
+		// STEP: freeze nodes in bucket
+		for _, n := range bucket {
+			node := clients[n]
+
+			nodeAddr, err := node.Addresses(ctx)
+			if err != nil {
+				return err
+			}
+
+			if len(nodeAddr.Underlay) == 0 {
+				return errors.New("not enough underlay addresses")
+			}
+
+			nodeIP := getIPFromUnderlays(nodeAddr.Underlay)
+			if nodeIP == "" {
+				continue
+			}
+
+			fmt.Printf("freezing block number for node %s ip %s\n", n, nodeIP)
+
+			cancelID, err := proxy.Execute(proxyClient.BlockNumberFreeze, nodeIP)
+			if err != nil {
+				return err
+			}
+
+			defer func(cancelID int) {
+				_ = proxy.Cancel(cancelID)
+			}(cancelID)
+		}
+
+		malfunctionEth += len(bucket)
```

Comment:
since some of the nodes in the bucket might be skipped (there's a `continue` statement), this count may be inconsistent